contracts/DelFiPrice.sol

```diff
@@ -62,7 +75,16 @@ contract DelFiPrice is OpenOracleView {
         }
 
         uint64[] memory sortedPrices = sort(postedPrices);
-        return sortedPrices[N / 2];
+
+        // if N is even, get the left and right medians and average them
+        if (N % 2 == 0) {
+            uint64 left = sortedPrices[(N / 2) - 1];
+            uint64 right = sortedPrices[N / 2];
+            return (left + right) / 2;
```

Comment:
needs overflow check. safemath and then cast? or inline?

Reply:
I think left is  n/2 and right n/2 +1 

n = 4
left = 4/2 = 2
right = 4/2 + 1 = 3

Reply:
n is length. so with a zero-indexed array with 4 objects, you'd actually want indexes 1 and 2.